common/data.go

```diff
@@ -58,6 +58,27 @@ func (f *DataFinalityState) UnmarshalYAML(unmarshal func(interface{}) error) err
 	return fmt.Errorf("invalid data finality state: %s", s)
 }
 
+// DataFinalityStateSlice provides custom YAML unmarshaling for []DataFinalityState
+type DataFinalityStateSlice []DataFinalityState
+
+func (f *DataFinalityStateSlice) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
Is this necessary? The DataFinalityState already has an UnmarshalYAML, isn't it working when accepted as array?